frontend/src/components/Course/Grading/GradingSession.tsx

```diff
@@ -0,0 +1,141 @@
+import { useGetExercisesOfSessionQuery, useUpdateExerciseMutation } from "../../../features/courses/exerciseApiSlice";
+import { useMemo, useState } from 'react';
+import GradingExercise from "./GradingExercise";
+import { useDispatch } from "react-redux";
+import { pushNotification } from "../../../features/notification/notificationSlice";
+import { useUpdateSessionMutation } from "../../../features/courses/sessionApiSlice";
+import { useUpdateTestMutation } from "../../../features/courses/testApiSlice";
+
+const GradingSession = (props: any) => {
+    const [exerciseGradeValues, setExerciseGradeValues] = useState<Record<number, string>>({})
+    const [testCoefficientValues, setTestCoefficientValues] = useState<Record<number, any>>({});
+    const [updateExercise] = useUpdateExerciseMutation();
+    const [updateSession] = useUpdateSessionMutation();
+    const [updateTest] = useUpdateTestMutation();
+    const dispatch = useDispatch();
+
+
+    const {
+        data: exercises,
+        isSuccess: exercisesIsSuccess,
+    } = useGetExercisesOfSessionQuery({ session_id: props.session?.id });
+
+    const sortedExercises = useMemo(() => {
+        if (exercisesIsSuccess) {
+            const exercisesToSort = structuredClone(exercises);
+            exercisesToSort.sort((a: any, b: any) => {
+                return a.title.localeCompare(b.title);
+            });
+            return exercisesToSort;
+        }
+        return exercises;
+    }, [exercises, exercisesIsSuccess]);
+
+    const handleSaveExerciseGrade = async (exercise: any) => {
+        if (exercise.grade)
+            await updateExercise(exercise)
```

Comment:
Maybe we could add error handling to this mutation ?

Reply:
This is not to handle errors but to avoid unnecessary updating of exercise notes that have not been updated.